oci/multi_region.go

```diff
@@ -384,3 +359,40 @@ func getCloudGuardConfiguration(ctx context.Context, d *plugin.QueryData, _ *plu
 
 	return response.Configuration, nil
 }
+
+// List out the regions supported by Oracle Cloud
+func listOciAvailableRegions(ctx context.Context, d *plugin.QueryData) ([]string, error) {
```

Comment:
@bigdatasourav If we can't get the list of regions from the API for any reason. e.g., permissions, blocked access to identity service, etc., what happens?

Also, below I can see:
```
endpointRegion := GetConfig(d.Connection).Regions[0]
```

Is this safe? What would this region usually be? Do we ever get different regions from the API response depending on what region we connect to?

Reply:
@cbruno10 A default user can access the list region API; no specific permission is required. 

We can call the list region API using any valid region in the client config, `Regions[0]` - here we use the first region.

Reply:
@bigdatasourav 
- Are there any ways for a user to be blocked on accessing the list regions API? It may be uncommon, but something we should account for if possible.
- In the AWS plugin we have default regions we fallback to for each partition in case no regions are defined. Do we need something like that here too?
- Are there any backward incompatible changes this PR introduces?

Reply:
- Any user in a particular tenancy can execute the API without specific permission. It is a default access to a user. I have confirmed the same with the OCI team as well. I have tested this with a user without any access. OCI also does not support creating `Deny` policies.
- We do not require a default region here, as if no regions are defined, the `listOciAvailableRegions` function will not be executed; the plugin will go ahead with the available default profile in `~/.oci/config`.
- There are no backward incompatible changes this PR introduces.